runtime/src/main/java/org/corfudb/runtime/collections/TxnContextShim.java

```diff
@@ -7,46 +7,59 @@
 import org.corfudb.runtime.exceptions.StaleRevisionUpdateException;
 import org.corfudb.runtime.object.transactions.AbstractTransactionalContext;
 import org.corfudb.runtime.object.transactions.TransactionalContext;
-import org.corfudb.runtime.view.ObjectsView;
-import org.corfudb.runtime.view.TableRegistry;
+import org.corfudb.runtime.view.Address;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 /**
- * Thin shim layer around CorfuStore's TxnContext for providing metadata management
- * capabilities.
+ * Shim layer around CorfuStore's TxnContext for providing metadata management
+ * capabilities, nested transaction support and more!
  *
  * Created by hisundar on 2020-09-16
  */
 @Slf4j
-public class TxnContextShim extends TxnContext implements AutoCloseable {
+public class TxnContextShim implements AutoCloseable {
```

Comment:
The Shim prefix for the name is not that great. Can we change it to a better name.

Reply:
ok, changed to ManagedTxnContext